src/Corporation/Actions.ts

```diff
@@ -332,8 +332,8 @@ export function buyMaterial(division: Division, material: Material, amt: number)
   if (!isRelevantMaterial(material.name, division)) {
     throw new Error(`${material.name} is not a relevant material for industry ${division.type}`);
   }
-  if (isNaN(amt) || amt < 0) {
-    throw new Error(`Invalid amount '${amt}' to buy material '${material.name}'`);
+  if (!Number.isFinite(amt) || amt < 0) {
+    throw new Error("Invalid value for amount field! Must be numeric and greater than or equal to 0");
```

Comment:
Let's keep the actual amount and material in the error, although specifying the proper constraints is good too.